src/bin/list_store.rs

```diff
@@ -0,0 +1,299 @@
+// "List Store" gtk3 demo ported to gtk-rs by Ömer Sinan Ağacan <omeragacan@gmail.com>
+
+extern crate gio;
+extern crate gtk;
+
+use gio::*;
+use gtk::*;
+
+use std::env::args;
+use std::rc::Rc;
+
+fn main() {
+    let application =
+        gtk::Application::new("com.github.gtk-rs.examples.list-store", Default::default())
+            .expect("Initialization failed...");
+
+    application.connect_activate(|app| {
```

Comment:
The correct code is:

```rust
    application.connect_startup(|app| {
        build_ui(app);
    });
application.connect_activate(|_| {});
```

`connect_activate` is fow when the same binary is launched again so generally, you just want to show the window, not create a new one.

Reply:
I actually copied from another example, but will upate.

Reply:
Best to also fix that example then :) See [here](https://github.com/sdroege/rustfest-rome18-gtk-gst-workshop/blob/a4a6cfd6fecd5485498a307991aefd6da8ca47cc/src/app.rs#L211) for a little bit cleaner pattern with all this, but not really needed for such a small example.